frontend/pages/reports/index.js

```diff
@@ -0,0 +1,227 @@
+import s from 'styled-components'
+import Router from 'next/router'
+import { useEffect, useState } from 'react'
+import renderPage from '../../renderPage.js'
+import { Icon, Empty, Checkbox, CheckboxLabel } from '../../components/common'
+import { doApiRequest, API_BASE_URL } from '../../utils'
+import { Container } from '../../components/common/Container'
+import { CLUBS_GREY } from '../../constants/colors'
+import Edit from './edit'
+
+const GroupLabel = s.h4`
+  margin-bottom: 0em !important;
+  font-size: 32px;
+  color: #626572;
+`
+
+const ColoredHeader = s.div`
+  background: linear-gradient(to right, #ef4c5f, #4954f4);
+  height: 7em;
+`
+
+const TransparentButton = s.button`
+  width: 190px;
+  height: 35px;
+  border-radius: 17px;
+  border: 0;
+  background: rgba(255,255,255,0.32);
+  float: right;
+  font-size: 15px;
+  font-family: Lato;
+  font-weight: 500;
+  font-stretch: normal;
+  font-style: normal;
+  line-height: normal;
+  letter-spacing: normal;
+  text-align: center;
+  color: rgba(255,255,255,1)
+`
+
+const serializeParams = params => {
+  return Object.keys(params)
+    .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}`)
+    .join('&')
+}
+
+const Reports = ({ nameToCode }) => {
+  const fields = {
+    Fields: Object.keys(nameToCode),
+  }
+
+  const [reports, setReports] = useState([])
+  const [reportFlag, updateReportFlag] = useState(false)
+
+  const [isEdit, setIsEdit] = useState(false)
+
+  useEffect(() => {
+    doApiRequest('/reports/?format=json')
+      .then(resp => {
+        if (resp.ok) {
+          return resp.json()
+        } else {
+          return []
+        }
+      })
+      .then(data => setReports(data))
+  }, [reportFlag])
+
+  const [includedFields, setIncludedFields] = useState(
+    (() => {
```

Comment:
you don't have to call the anonymous function manually here, useState accepts functions!